libs/chonky/src/lib.rs

```diff
@@ -1,26 +1,134 @@
 #![doc = include_str!("../README.md")]
 
-/// Adds two numbers together
-///
-/// # Example
-///
-/// ```rust
-/// use chonky::add;
-///
-/// assert_eq!(add(1, 3), 4);
-/// ```
-#[must_use]
-pub const fn add(left: u64, right: u64) -> u64 {
-    left + right
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ChonkyError {
+    #[error("parsing error in pdf")]
+    ReadPdf,
+    #[error("pdfium error")]
+    Pdfium,
+    #[error("write error to system")]
+    Write,
+    #[error("Issues with CLI input")]
+    Arguments,
+}
+
+pub mod pdf_segmentation {
+    use error_stack::{Report, ResultExt};
+    use image::DynamicImage;
+    use pdfium_render::prelude::*;
+
+    use crate::ChonkyError;
+
+    /// Function to read the pdf
+    ///
+    /// # Errors
+    ///
+    /// Will return `Err` if `filename` does not exist or the user does not have
+    /// permission to read it.
+    pub fn load_pdf<'a>(
+        pdfium: &'a Pdfium,
+        file_path: &str,
+    ) -> Result<PdfDocument<'a>, Report<ChonkyError>> {
+        pdfium
+            .load_pdf_from_file(file_path, None)
+            .map_err(|err| Report::new(err).change_context(ChonkyError::Pdfium))
+    }
+
+    /// Takes in a pdf document and returns a vector list where each page
+    /// is processed into a raw image that can be later converted to any image format
+    ///
+    /// # Errors
+    /// Return an `PdfiumError::ImageError` if there was a problem with the image processing
+    /// operation, occurs when image cannot be encoded into specific image format
```

Comment:
It's possible to cross-link documentation ([intra-doc links](https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html)). There is a quite easy rule: If it's in the module scope (e.g. as `DynamicImage`, it can be used directly, otherwise you can link it like it would be a markdown link. I.e.
```suggestion
    /// # Errors
    ///
    /// Return an [`PdfiumError::ImageError`] if there was a problem with the image processing
    /// operation, occurs when image cannot be encoded into specific image format
```
or
```suggestion
    /// # Errors
    ///
    /// Return an [`ImageError`] if there was a problem with the image processing
    /// operation, occurs when image cannot be encoded into specific image format
    ///
    /// [`ImageError`]: PdfiumError::ImageError
```

If a link is not valid, the CI will complain.